### PR TITLE
BGDIINF_SB-1391: adding favorite icon support

### PR DESCRIPTION
**Pull request by ltkum**
Issue : When looking at pages from service-stac through a browser, there
was no favorite icons.

Fix : Uploaded a favorite Icon in the project in app/stac_api/static/
 In settings, told django that the /static/ directory is where to find
static content
 In urls, added an url redirection for the favicon (since it would be at
/static/favicon.ico rather than /favicon.ico as most browsers think it
is.)
 In templates, specified that the favicon is in the static directory,
for the browsers that would not follow redirections for favicons

Issue : 
  Endpoints that do not rely on templates (for example : json responses meant to be called via api_calls, such as /checker for the most abvious example) will not display a favicon. Since these are not meant to be called via a browser, I do not think it matters much but I think it's faire to mention it.

### CODE REVIEW

File: app/config/settings_prod.py [hunk 1]
```diff
@@ -274,3 +274,7 @@ def get_logging_config():
 # By default django_promtheus tracks the number of migrations
 # This causes troubles in various places so we disable it
 PROMETHEUS_EXPORT_MIGRATIONS = False
+
+
+# Where the application goes to find its static files.
+STATIC_URL = "/static/"
```
Comment (ltshb) [inline app/config/settings_prod.py:280]: The `STATIC_URL` is already defined in line 178 and has been modified by  PR #192

File: app/stac_api/urls.py [hunk 2]
```diff
@@ -45,4 +47,8 @@
         AssetDetail.as_view(),
         name='asset-detail'
     ),
+    path(
+        "favicon.ico",
+        RedirectView.as_view(url=staticfiles_storage.url("favicon.ico")),
+    ),
```
Comment (ltshb) [inline app/stac_api/urls.py:53]: This should not be needed and the static file should be served by Whitenoise.

**Review by ltshb — CHANGES_REQUESTED** [review]
I would put the favicon in `app/stac_api/templates` as this path is already part of the static asset served by Whitenoise.

File: app/templates/admin/base.html [hunk 3]
```diff
@@ -1,4 +1,7 @@
 {% extends "admin/base.html" %}
+{% load static %}
+
+<link rel="icon"href="{% static 'favicon.ico' %}">
```
Comment (boecklic) [inline app/templates/admin/base.html:4]: In Django it's usually a good idea to place everything inside a block/extend a block. For the favicon you can use the block `extrahead` (see https://stackoverflow.com/a/64368855/9896222)
(plus, there's a missing space between 'icon' and href)

File: app/templates/rest_framework/base.html [hunk 4]
```diff
@@ -1,4 +1,7 @@
 {% extends "rest_framework/base.html" %}
+{% load static %}
+
+<link rel="icon"href="{% static 'favicon.ico' %}">
```
Comment (boecklic) [inline app/templates/rest_framework/base.html:4]: dito

File: app/stac_api/templates/admin/ol_swisstopo.html [hunk 5]
```diff
@@ -1,3 +1,7 @@
+{% load static %}
+
+<link rel="icon"href="{% static 'favicon.ico' %}">
```
Comment (boecklic) [inline app/stac_api/templates/admin/ol_swisstopo.html:3]: dito

**Review by boecklic — CHANGES_REQUESTED** [review]

File: app/config/settings_prod.py (hunk 1 above)
Comment (ltkum) [reply app/config/settings_prod.py]: and now we've removed it.

File: app/stac_api/urls.py (hunk 2 above)
Comment (ltkum) [reply app/stac_api/urls.py]: I confirm this is not needed with Whitenoise.

File: app/templates/admin/base.html (hunk 3 above)
Comment (ltkum) [reply app/templates/admin/base.html]: It's been corrected in all files.

**Review by ltshb — CHANGES_REQUESTED** [review]
Favicon on the landing page is not working, it only works for django admin page.

**Comment by ltkum** [comment]
image

@ltshb  It seems to be working for me locally. Is there anything different between your environment and mine ?

**Comment by ltkum** [comment]
From what I could see, it appears in firefox but not in chromium. 
Chromium doesn't even try to get it, while firefox tries to get it (twice : once at /favicon.ico, once at /api/stac/static/favicon.ico)
The head link to favicon is indeed not there. I'll look into it.

**Comment by ltkum** [comment]
From what I understand : 
I'll have to check how to add a <head> element to class based views with no templates.

**Comment by ltshb** [comment]
It looks like it search for the favicon at the wrong place, the browser does a GET	http://127.0.0.1:8000/favicon.ico ! I used firefox  in private mode, but I don't think this is related to firefox.

**Review by ltshb — APPROVED** [review]
:+1:

**Review by boecklic — APPROVED** [review]
:+1:

**Review by rebert — COMMENTED** [review]
@ltkum nice. I wanted to have an eye on the icon, but did not find it. Where is the famous icon located?

**Review by rebert — APPROVED** [review]

**Pull request merged**